compiler/ast/src/types/type_.rs

```diff
@@ -51,6 +51,7 @@ impl Type {
     ///
     /// Flattens array syntax: `[[u8; 1]; 2] == [u8; (2, 1)] == true`
     ///
+    // TODO: Does not seem to flatten?
```

Comment:
You are correct.
The array type is not implemented so we should probably delete this method for now.